client/gutenberg/extensions/mailchimp/edit.jsx

```diff
@@ -1,44 +1,239 @@
+		const { setAttributes } = this.props;
+		setAttributes( { errorLabel } );
+		this.auditionNotification( NOTIFICATION_ERROR );
+	};
+
+	updateEmailPlaceholder = emailPlaceholder => {
+		const { setAttributes } = this.props;
+		setAttributes( { emailPlaceholder } );
+		this.clearAudition();
+	};
+
+	updateSubmitLabel = submitLabel => {
+		const { setAttributes } = this.props;
+		setAttributes( { submitLabel } );
+		this.clearAudition();
+	};
+
+	render = () => {
+		const { attributes, className, notices, noticeUI, setAttributes } = this.props;
+		const { audition, connected, connectURL } = this.state;
+		const {
+			emailPlaceholder,
+			title,
+			submitLabel,
+			consentText,
+			processingLabel,
+			successLabel,
+			errorLabel,
+		} = attributes;
+		const classPrefix = 'wp-block-jetpack-mailchimp-';
+		const waiting = (
+			<Placeholder icon={ icon } notices={ notices }>
+				<Spinner />
+			</Placeholder>
+		);
+		const placeholder = (
+			<Placeholder icon={ icon } label={ __( 'Mailchimp' ) } notices={ notices }>
+				<div className="components-placeholder__instructions">
+					{ __(
+						'You need to connect your MailChimp account and choose a list in order to start collecting Email subscribers.'
+					) }
+					<br />
+					<ExternalLink href={ connectURL }>{ __( 'Set up MailChimp form' ) }</ExternalLink>
+					<br />
+					<br />
+					<Button isDefault onClick={ this.apiCall }>
+						{ __( 'Re-check Connection' ) }
+					</Button>
+				</div>
+			</Placeholder>
+		);
+		const inspectorControls = (
+			<InspectorControls>
+				<PanelBody title={ __( 'Text Elements' ) }>
+					<TextControl
+						label={ __( 'Email Placeholder' ) }
+						value={ emailPlaceholder }
+						onChange={ this.updateEmailPlaceholder }
+					/>
+					<TextControl
+						label={ __( 'Submit button label' ) }
+						value={ submitLabel }
+						onChange={ this.updateSubmitLabel }
+					/>
+				</PanelBody>
+				<PanelBody title={ __( 'Notifications' ) }>
+					<TextControl
+						label={ __( 'Processing text' ) }
+						value={ processingLabel }
+						onChange={ this.updateProcessingText }
+					/>
+					<TextControl
+						label={ __( 'Success text' ) }
+						value={ successLabel }
+						onChange={ this.updateSuccessText }
+					/>
+					<TextControl
+						label={ __( 'Error text' ) }
+						value={ errorLabel }
+						onChange={ this.updateErrorText }
+					/>
+				</PanelBody>
+				<PanelBody title={ __( 'Mailchimp Connection' ) }>
+					<ExternalLink href={ connectURL }>{ __( 'Manage Connection' ) }</ExternalLink>
+				</PanelBody>
+			</InspectorControls>
+		);
+		const blockContent = (
+			<div className={ className }>
+				<RichText
+					tagName="h3"
+					placeholder={ __( 'Write title' ) }
+					value={ title }
+					onChange={ value => setAttributes( { title: value } ) }
+					inlineToolbar
+				/>
+				{ ! audition && (
+					<form ref={ this.formRef }>
+						<TextControl placeholder={ emailPlaceholder } onChange={ () => false } />
```

Comment:
NIT (no change necessary): It might be preferable to use lodash's `noop` for this, which we get "for free" in Gutenberg, and it will be constant across rerenders.